### PR TITLE
net: added ng_netdump module

### PR DESCRIPTION
**Pull request by haukepetersen**
this module simply dumps received packets to STDOUT. The implementation is meant as a base for future extensions that could print more useful stuff...

usage:
 - initialize pktdump, save the PID
 - register that PID with netreg for whatever type of packets you are interested in

### CODE REVIEW

File: sys/net/crosslayer/ng_pktdump/ng_pktdump.c [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+void _dump(ng_pktsnip_t *pkt)
+{
+    int snips = 0;
+    int size = 0;
+
+    while (pkt != NULL) {
+        printf("~~ SNIP %2i - size: %3i byte, type: ", snips, pkt->size);
+        _dump_type(pkt->type);
+        puts("");
```
Comment (authmillenon) [inline sys/net/crosslayer/ng_pktdump/ng_pktdump.c]: You can use the `od` model to also dump the packets content. Or should this happen in a more structured way in `_dump_type`?

Comment (haukepetersen) [inline sys/net/crosslayer/ng_pktdump/ng_pktdump.c]: Using the `od` module would be a good idea. How about merging this as is in the simple form and improving it then?

Comment (authmillenon) [inline sys/net/crosslayer/ng_pktdump/ng_pktdump.c]: +1

File: sys/include/net/ng_pktdump.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+extern "C" {
+#endif
+
+/**
+ * @brief   Start the packet dump thread and listening for incoming packets
+ *
+ * @param stack[in]         stack for the packet dump thread
+ * @param stacksize[in]     size of @p stack
+ * @param priority[in]      priority of the packet dump thread
+ * @param name[in]          name for the packet dump thread
```
Comment (authmillenon) [inline sys/include/net/ng_pktdump.h]: `s/@param \([^[]\+\)[in]/@param[in] \1/`. This is why Travis is grumpy.

**Comment by authmillenon** [comment]
Apart from the wrong possitioning of the parameter directions: ACK.

File: sys/net/crosslayer/ng_pktdump/ng_pktdump.c [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            case NG_NETAPI_MSG_TYPE_RCV:
+                puts("PKTDUMP: data received:\n");
+                _dump((ng_pktsnip_t *)msg.content.ptr);
+                break;
+            case NG_NETAPI_MSG_TYPE_SND:
+                puts("PKTDUMP: data to send:\n");
+                _dump((ng_pktsnip_t *)msg.content.ptr);
+                break;
+            default:
+                puts("PKTDUMP: received something unexpected\n");
```
Comment (authmillenon) [inline sys/net/crosslayer/ng_pktdump/ng_pktdump.c]: When I merge this branch into my branch `tmp2` I get this message when I put in `ifconfig`, with this module registered to ICMPv6.

Comment (authmillenon) [inline sys/net/crosslayer/ng_pktdump/ng_pktdump.c]: (works like a charm :-))

File: sys/net/crosslayer/ng_pktdump/ng_pktdump.c [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+{
+    (void)arg;
+    msg_t msg;
+
+    while (1) {
+        msg_receive(&msg);
+
+        switch (msg.type) {
+            case NG_NETAPI_MSG_TYPE_RCV:
+                puts("PKTDUMP: data received:\n");
```
Comment (authmillenon) [inline sys/net/crosslayer/ng_pktdump/ng_pktdump.c]: `s/\n//`

File: sys/net/crosslayer/ng_pktdump/ng_pktdump.c [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                _dump((ng_pktsnip_t *)msg.content.ptr);
+                break;
+            case NG_NETAPI_MSG_TYPE_SND:
+                puts("PKTDUMP: data to send:\n");
```
Comment (authmillenon) [inline sys/net/crosslayer/ng_pktdump/ng_pktdump.c]: `s/\n//`

File: sys/net/crosslayer/ng_pktdump/ng_pktdump.c (hunk 3 above)
Comment (authmillenon) [inline sys/net/crosslayer/ng_pktdump/ng_pktdump.c]: `s/\n//` however ;-)

File: sys/include/net/ng_pktdump.h (hunk 2 above)
Comment (haukepetersen) [inline sys/include/net/ng_pktdump.h]: this is why it looked so funny... :-)

**Comment by haukepetersen** [comment]
addressed comments.

**Comment by authmillenon** [comment]
ACK upholds, please squash

**Comment by haukepetersen** [comment]
squashed, waiting for Travis.

**Pull request merged**